File: distributivo-frontend/src/app/feature/subject/subject.ts
export interface SubjectSelect {
    id:   string;
    name: string;
}

export interface Subject {
    id:               string;
    code:             string;
    name:             string;
    theoreticalHours: string;
    laboratoryHours:  string;
    career:           number;
}

export interface Paginable {
    total:      number;
    page:       number;
    totalPages: number;
    capacity:   number;
    subjects:   any[];
}

File: distributivo-backend/src/main/java/edu/yavirac/distributivobackend/feature/subject/Subject.java
package edu.yavirac.distributivobackend.feature.subject;


import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import lombok.Data;

@Data
@Table("subject")
public class Subject {
    @Id
    @Column("id")
    private long id;
    private String code;
    private String name;
    @Column("theoretical_hours")
    private Number theoreticalHours;
    @Column("laboratory_hours")
    private Number laboratoryHours;
    private Boolean status;
    private Number career;

}
